jina/executors/indexers/vector.py

```diff
@@ -240,8 +240,28 @@ def __init__(self, metric: str = 'euclidean',
         self.metric = metric
         self.backend = backend
 
+    @staticmethod
+    def _find_smallest_distances(dist: 'np.array', top_k: int) -> Tuple['np.ndarray', 'np.ndarray']:
+        """ Find top-k smallest distances in ascending order.
+
+        Idea is to use partial sort to retrieve top-k smallest distances unsorted and then sort these
+        in ascending order. Equivalent to full sort but faster for n >> k. If n == k revert to full sort.
+
+        """
+        if top_k == dist.shape[1]:
```

Comment:
changing to `top_k >= dist.shape[1]` should make it more clear that it works if `top_k` is greater than the indexed documents.

Reply:
Very good point, will add this and also add test case